##Atividade de Cálculo Numérico

In [ ]:
import math
math.sqrt(2)


1.4142135623730951

**Questão 1: Esse cálculo pode ser considerado um exemplo de cálculo direto? Justifique com base na definição apresentada em aula.**

Sim, pois de acordo com a definição normalmente apresentada em aula, cálculo direto é aquele em que o resultado é obtido imediatamente por meio de uma expressão ou função já definida, sem necessidade de estruturas condicionais, repetições ou etapas intermediárias complexas.

A função math.sqrt(2) calcula diretamente a raiz quadrada de 2 utilizando uma função pronta da biblioteca math. O valor é retornado de forma imediata, sem depender de outras variáveis ou processos adicionais.
Portanto, trata-se de um cálculo direto, pois o resultado é obtido diretamente pela aplicação de uma função matemática a um valor específico.

In [ ]:
x = 1

for i in range(5):
    x = (x + 2/x) / 2

print(x)

1.414213562373095


**Questão 2: O valor obtido melhora a cada iteração?**

Sim. O valor de x melhora a cada iteração porque o algoritmo utiliza o método de Newton-Raphson para aproximar √2.

A cada repetição, o novo valor de x fica mais próximo da raiz quadrada real de 2 (≈ 1,41421356...). Ou seja, o erro diminui progressivamente, tornando a aproximação cada vez mais precisa.








**Questão 2: Esse processo pode ser classificado como
cálculo direto ou iterativo? Explique.**

Esse processo é classificado como cálculo iterativo. O resultado não é obtido em uma única operação ou fórmula aplicada diretamente ao número 2.

Em vez disso, o valor é refinado ao longo de várias repetições dentro do laço (for), onde cada novo valor depende do anterior. Assim, trata-se de um método que produz aproximações sucessivas até alcançar um valor suficientemente próximo da solução desejada.



In [ ]:
print(round(3.1415926535, 2))
print(round(3.1415926535, 4))

3.14
3.1416


**Questão 3: Qual dos valores é mais correto?**

O valor 3.1416 numericamente mais preciso, pois possui mais casas decimais e, portanto, está mais próximo do valor real de π.

**Questão 3: A resposta depende do problema considerado?
Justifique.**
Sim, depende do problema considerado.

A quantidade de casas decimais necessárias está relacionada ao nível de precisão exigido.  
- Em cálculos simples ou situações do dia a dia, 3.14 pode ser suficiente.  
- Em cálculos científicos ou de engenharia, pode ser necessário usar mais casas decimais, como 3.1416 ou ainda mais.

Portanto, o valor “mais correto” é aquele que atende ao grau de precisão exigido pelo contexto do problema.